src/drivers/intel/cavs/ssp.c

```diff
@@ -365,10 +366,38 @@ static int ssp_set_config(struct dai *dai,
 		if (ssp_freq[clk_index].enc != CLOCK_SSP_XTAL_OSCILLATOR)
 			sscr0 |= SSCR0_ECS;
 	} else {
-		trace_ssp_error("ssp_set_config() error: BCLK %d",
-				config->ssp.bclk_rate);
-		ret = -EINVAL;
-		goto out;
+		/* try with M/N */
+		for (i = MAX_SSP_FREQ_INDEX; i >= 0; i--) {
+			/* M/N cannot be greater than 1/2 */
```

Comment:
this is no longer valid

Reply:
@abonislawski I guess you mean that not the M/N has to be <= 1/2, but factor overall after applying SCR also, right?

Reply:
@jajanusz this is probably how it works because mdiv is applied first so the input for mndivider is selected freq / mdiv so in this case you should add mdiv to this check. But this is only assumption, no one clearly confirmed it.
According to spec this check is not valid too, spec says only about M/N <1/2 but its not valid for us too